server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java

```diff
@@ -311,6 +254,14 @@ public DateFieldType(String name) {
             this(name, true, true, DEFAULT_DATE_TIME_FORMATTER, Resolution.MILLISECONDS, Collections.emptyMap());
         }
 
+        public DateFieldType(String name, DateFormatter dateFormatter) {
```

Comment:
Do you think it'd be cleaner to use the builder to get these in tests?

Reply:
Builders give us field mappers though, not field types, so it feels like an extra unnecessary step.

Reply:
Yeah. I can see that. I'm just not a big fan of having a bunch of ctors, especially when we have a fancy builder system already that makes it super clear what you are doing. Oh well. Its cool.